src/saveload/saveload.cpp

```diff
@@ -1217,6 +1217,67 @@ static void *IntToReference(size_t index, SLRefType rt)
 	}
 }
 
+/**
+ * Template class to help with list-like types.
+ */
+template <template<typename, typename> typename Tstorage, typename Tvar, typename Tallocator = std::allocator<Tvar>>
+class SlStorageHelper {
+	typedef Tstorage<Tvar, Tallocator> SlStorageT;
+public:
+	/**
+	 * Internal templated helper to return the size in bytes of a list-like type.
+	 * @param storage The storage to find the size of
+	 * @param conv VarType type of variable that is used for calculating the size
+	 */
+	static size_t SlCalcLen(const void *storage, VarType conv)
+	{
+		const SlStorageT *l = (const SlStorageT *)storage;
+
+		int type_size = 4;
+		int item_size = SlCalcConvFileLen(conv);
+		return l->size() * item_size + type_size;
+	}
+
+	/**
+	 * Internal templated helper to save/load a list-like type.
+	 * @param storage The storage being manipulated.
+	 * @param conv VarType type of variable that is used for calculating the size.
+	*/
+	static void SlSaveLoad(void *storage, VarType conv)
+	{
+		SlStorageT *list = (SlStorageT *)storage;
+
+		switch (_sl.action) {
+			case SLA_SAVE: {
```

Comment:
Unneeded brackets, especially as you remove them in the next commit.

Reply:
So they are removed in the next commit :)

Again, moved code :D Not going to resolve this one, as you pointed out, fixed in next commit :)